Content.Shared/ChangeNameInContainer/ChangeNameInContainerSystem.cs

```diff
@@ -0,0 +1,28 @@
+using Content.Shared.Chat;
+using Robust.Shared.Containers;
+using Content.Shared.Whitelist;
+using Content.Shared.Speech;
+
+namespace Content.Shared.ChangeNameInContainer;
+public sealed partial class ChangeNameInContainerSystem : EntitySystem
+{
+    [Dependency] private readonly SharedContainerSystem _container = default!;
+    [Dependency] private readonly EntityWhitelistSystem _whitelist = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<ChangeVoiceInContainerComponent, TransformSpeakerNameEvent>(UpdateName);
+    }
+
+    private void UpdateName(Entity<ChangeVoiceInContainerComponent> entity, ref TransformSpeakerNameEvent args)
```

Comment:
```suggestion
    private void OnTransferSpeakerName(Entity<ChangeVoiceInContainerComponent> ent, ref TransformSpeakerNameEvent args)
```
naming convention
use ent below as well